examples/minimal_agent_production/steps/ingest.py

```diff
@@ -5,70 +5,160 @@
+    # Determine input source and extract content
+    document_content = ""
+    actual_filename = filename
+
+    if content:
+        document_content = content
+        if not actual_filename:
+            actual_filename = "direct_input.txt"
+    elif url:
+        document_content = _download_from_url(url)
+        if not actual_filename:
+            # Extract filename from URL
+            parsed_url = urlparse(url)
+            actual_filename = (
+                os.path.basename(parsed_url.path) or "downloaded_content.txt"
+            )
+    elif path:
+        document_content = _load_from_path(path)
+        if not actual_filename:
+            actual_filename = os.path.basename(path)
+    else:
+        error_msg = (
+            "No content source provided. Must specify content, url, or path."
+        )
         raise ValueError(error_msg)
 
-    if not filename:
-        error_msg = "Filename cannot be empty"
-        logger.error(error_msg)
+    # Validate content
+    if not document_content or not document_content.strip():
+        error_msg = (
+            f"Document content is empty after processing: {actual_filename}"
+        )
         raise ValueError(error_msg)
 
-    # Log document statistics for monitoring
-    content_stats = {
-        "character_count": len(content),
-        "word_count": len(content.split()),
-        "line_count": len(content.split("\n")),
-        "document_type": document_type,
-        "analysis_type": analysis_type,
-    }
-
-    logger.info(f"Document statistics: {content_stats}")
-
     # Create and return the document request
     document = DocumentRequest(
-        filename=filename,
-        content=content,
+        filename=actual_filename,
+        content=document_content,
         document_type=document_type,
-        analysis_type=analysis_type,
     )
 
-    logger.info(f"Successfully ingested document: {filename}")
     return document
+
+
+def _download_from_url(url: str) -> str:
+    """Download content from a URL.
+
+    Args:
+        url: URL to download from
+
+    Returns:
+        Downloaded content as string
+
+    Raises:
+        ValueError: If download fails or content is invalid
+    """
+    try:
+        response = requests.get(url, timeout=30)
+        response.raise_for_status()
+
+        # Handle different content types
+        content_type = response.headers.get("content-type", "").lower()
+
+        if "text" in content_type or "json" in content_type:
+            return response.text
+        elif "html" in content_type:
+            # Basic HTML text extraction (you might want to use BeautifulSoup for better parsing)
+            import re
+
+            # Simple HTML tag removal
+            clean_text = re.sub(r"<[^>]+>", "", response.text)
+            return clean_text
+        else:
+            # For binary content, return a placeholder
+            return f"[Binary content downloaded from {url}, {len(response.content)} bytes]"
```

Comment:
You have to do the HTML check first as it's really `text/html` and otherwise
it'd never run.
```suggestion

        if "html" in content_type:
            # Basic HTML text extraction (you might want to use BeautifulSoup for better parsing)
            import re

            # Simple HTML tag removal
            clean_text = re.sub(r"<[^>]+>", "", response.text)
            return clean_text
        elif "text" in content_type or "json" in content_type:
            return response.text
        else:
            # For binary content, return a placeholder
            return f"[Binary content downloaded from {url}, {len(response.content)} bytes]"
```